src/use_promise_guard/index.es

```diff
@@ -0,0 +1,19 @@
+import useMounted from '../use_mounted';
+
+const usePromiseGuard = (deps) => {
+  const isMounted = useMounted(deps);
+
+  const execute = (getPromise) => new Promise((resolve, reject) => {
+    getPromise(resolve, reject)
```

Comment:
why do you pass in resolve and reject from the promise you are just creating? how would you call this thing?

Reply:
i passed resolve and reject to getPromise to allow immediate bail out instead of returning Promise.resolve(). I suppose this is an overkill.

Reply:
What would be a use case?